src/Raven.Server/Documents/Sharding/Subscriptions/ShardedSubscriptionWorker.cs

```diff
@@ -192,18 +216,37 @@ protected override async Task TrySetRedirectNodeOnConnectToServerAsync()
         {
             if (_databaseContext.ShardsTopology.TryGetValue(_shardNumber, out var topology))
             {
-                var node = topology.WhoseTaskIsIt(_databaseContext.ServerStore.Engine.CurrentState, _state.SubscriptionState, null);
-                if (node == null || _shardRequestExecutor == null)
+                var node = topology.WhoseTaskIsIt(RachisState.Follower, _state.SubscriptionState, null);
```

Comment:
why follower?

Reply:
I don't want to break connection in case the cluster is voting, thats why I pass the state manually (we do that in all `WhoseTaskIsIt` for subscription on server side)